src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -293,14 +293,18 @@ def _write_tsv(self, output: IO[str]) -> None:
 
     def _create_bdbag_archive(self) -> str:
         with TemporaryDirectory() as bag_path:
-            bdbag_api.make_bag(bag_path)
-            with open(os.path.join(bag_path, 'data', 'samples.tsv'), 'w') as samples_tsv:
+            manifest_folder = os.path.join(bag_path, 'manifest')
+            os.makedirs(manifest_folder)
+            bdbag_api.make_bag(manifest_folder)
+            with open(os.path.join(manifest_folder, 'data', 'samples.tsv'), 'w') as samples_tsv:
                 self._write_bdbag_samples_tsv(samples_tsv)
-            bag = bdbag_api.make_bag(bag_path, update=True)  # update TSV checksums
-            assert bdbag_api.is_bag(bag_path)
-            bdbag_api.validate_bag(bag_path)
+            bag = bdbag_api.make_bag(manifest_folder, update=True)  # update TSV checksums
+            assert bdbag_api.is_bag(manifest_folder)
+            bdbag_api.validate_bag(manifest_folder)
             assert bdbag_api.check_payload_consistency(bag)
-            return bdbag_api.archive_bag(bag_path, 'zip')
+            new_zip_path = os.path.join(os.path.dirname(bag_path), 'manifest.zip')
```

Comment:
#991 does not specify this change. What's the reason for changing the ZIP file's name?

Reply:
The ZIP file isn't being renamed. I'm using the `os.rename()` to move the ZIP file out of the temporary folder. `bdbag_api.archive_bag()` creates the ZIP file in the temporary directory because `bag_path` refers to a subfolder within the temporary folder. Since the temporary folder is deleted before anything is uploaded into the S3 bucket, I need to move that ZIP file out of the temporary folder.